connectors/overview.mdx

```diff
@@ -139,3 +139,27 @@ able to read data.
 
 To address this, source tasks with no work assigned to them will mark themselves as idle
 immediately.
+
+## Environment variable substitution
+
+Some fields in the connection configurations can use environment variables to substitute values
```

Comment:
Is there a way we can document which fields support variable substitution?

Reply:
In the UI all fields that support substitution automatically say so in the description - 
![image](https://github.com/ArroyoSystems/arroyo-docs/assets/8881183/0be94407-a45c-4783-89b6-fd8efb652bea)

Think we should mention here that it's controlled by the `format` attribute in the json schema?

Reply:
Might be fine to just list them for the moment.

Reply:
I've added the line
> Currently, environment variable substitution is supported for fields containing API keys,
usernames, passwords, and HTTP headers.